server/src/main/java/org/opensearch/index/translog/TranslogReader.java

```diff
@@ -129,8 +129,9 @@ public int totalOperations() {
         return totalOperations;
     }
 
+    // we can provide an isBetween method here if we don't want to change visibility on this method.
     @Override
-    final Checkpoint getCheckpoint() {
+    public final Checkpoint getCheckpoint() {
```

Comment:
not totally in love with this, feel like it was package private for a reason.  I think an isBetween method here may be more appropriate.

Reply:
Can't we keep it package private and have the plugin implement this in the same package (e.g., `o.opensearch.index.translog.ReplicationTranslogDeletionPolicy`) only exposing the public facing APIs needed for the plugin specific consumer classes?

Reply:
ah yep, done.